src/app/codegen-data-model-provider/CodegenDataModelProvider_Write.cpp

```diff
@@ -51,7 +51,7 @@ using Protocols::InteractionModel::Status;
 ///
 /// If it returns std::nullopt, then there is no AAI to handle the given path
 /// and processing should figure out the value otherwise (generally from other ember data)
-std::optional<CHIP_ERROR> TryWriteViaAccessInterface(const ConcreteAttributePath & path, AttributeAccessInterface * aai,
+std::optional<CHIP_ERROR> TryWriteViaAccessInterface(const ConcreteDataAttributePath & path, AttributeAccessInterface * aai,
```

Comment:
This was a quite non-obvious bug: Data contains `mListOp` however without it, there is no such thing. And amazingly enough they are cross-assignable so this was failing silently.

I did not add  an explicit unit test here because TestWriteChunking catches this big time.